parsers/ES_CN.py

```diff
@@ -6,40 +6,37 @@
 
 
 def fetch_consumption(country_code='ES-CN', session=None):
-    if not session:
-        session = Session
+    ses = session or session()
 
     try:
-        response = CanaryIslands(session).get()
+        response = CanaryIslands(ses).get()
     except NoDataException:
         response = None
     except TimestampException:
         response = None
 
     if not response:
         datetime = utcnow().datetime
-        consumption = None
     else:
         datetime = get(response.timestamp).datetime
-        consumption = response.demand
 
     data = {
         'countryCode': country_code,
         'datetime': datetime,
-        'consumption': consumption,
         'source': 'demanda.ree.es'
     }
 
+    if response:
```

Comment:
If there's no response you should return None instead of returning `data` without the `consumption` key.
I suggest return data inside the if, and returning None in an additional else.